src/operator/controllers/poduserpassword/db_credentials_pod_reconciler.go

```diff
@@ -79,57 +94,103 @@ func (e *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, errors.Wrap(err)
 	}
-
 	e.recorder.Event(&pod, v1.EventTypeNormal, ReasonEnsuredPodUserAndPassword, "Ensured user-password credentials in specified secret")
 	return ctrl.Result{}, nil
 }
 
-func (e *Reconciler) ensurePodUserAndPasswordPostgresSecret(ctx context.Context, pod *v1.Pod, serviceName string, secretName string) error {
+func (e *Reconciler) ensurePodUserAndPasswordSecret(ctx context.Context, pod *v1.Pod, secretName string) (error, string) {
 	log := logrus.WithFields(logrus.Fields{"pod": pod.Name, "namespace": pod.Namespace})
-	err := e.client.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: secretName}, &v1.Secret{})
+	secret := v1.Secret{}
+	err := e.client.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: secretName}, &secret)
 	if apierrors.IsNotFound(err) {
 		log.Debug("Creating user-password credentials secret for pod")
 		password, err := createServicePassword()
 		if err != nil {
-			return errors.Wrap(err)
-		}
-		clusterID, err := clusterutils.GetClusterUID(ctx)
-		if err != nil {
-			return errors.Wrap(err)
+			return errors.Wrap(err), ""
 		}
 
-		username := clusterutils.BuildHashedUsername(serviceName, pod.Namespace, clusterID)
-		pgUsername := clusterutils.KubernetesToPostgresName(username)
+		databaseUsername := pod.Annotations[databaseconfigurator.DatabaseUsernameAnnotation]
 
-		secret := buildUserAndPasswordCredentialsSecret(secretName, pod.Namespace, pgUsername, password)
+		secret := buildUserAndPasswordCredentialsSecret(secretName, pod.Namespace, databaseUsername, password)
 		log.WithField("secret", secretName).Debug("Creating new secret with user-password credentials")
 		if err := e.client.Create(ctx, secret); err != nil {
-			return errors.Wrap(err)
+			return errors.Wrap(err), ""
 		}
-		return nil
+		return nil, password
 	}
 
 	if err != nil {
-		return errors.Wrap(err)
+		return errors.Wrap(err), ""
 	}
 	log.Debug("Secret exists, nothing to do")
+	return nil, string(secret.Data["password"])
+}
+
+func (e *Reconciler) ensurePasswordInDatabases(ctx context.Context, pod v1.Pod, password string) error {
+	databases := strings.Split(pod.Annotations[databaseconfigurator.DatabaseAccessAnnotation], ",")
+	username := pod.Annotations[databaseconfigurator.DatabaseUsernameAnnotation]
+	pgServerConfigs := otterizev1alpha3.PostgreSQLServerConfigList{}
+	err := e.client.List(ctx, &pgServerConfigs)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+	mysqlServerConfigs := otterizev1alpha3.MySQLServerConfigList{}
+	err = e.client.List(ctx, &mysqlServerConfigs)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+
+	for _, database := range databases {
+		var dbconfigurator databaseconfigurator.DatabaseConfigurator
+		mysqlConf, found := lo.Find(mysqlServerConfigs.Items, func(config otterizev1alpha3.MySQLServerConfig) bool {
+			return config.Name == database
+		})
+		if found {
+			dbconfigurator, err = mysql.NewMySQLConfigurator(ctx, mysqlConf.Spec)
```

Comment:
Don't forget to close DB connections (you should `defer dbconfigurator.Close()` somewhere, probably after splitting the code into a dedicated function).